grails-app/conf/UrlMappings.groovy

```diff
@@ -101,6 +101,11 @@ class UrlMappings {
             controller = { "productApi" }
         }
 
+        "/api/products/$productCode/inventoryItems/$lotNumber"(parseRequest: true) {
```

Comment:
Sending the ID instead of productCode would save us avoid an extra query on the database.